tests/deployments/puppet/puppet_test.py

```diff
@@ -87,52 +97,57 @@ def run_puppet_agent(cont, backend, monitors, agent_version=None, stage=STAGE):
     assert code in (0, 2), output.decode("utf-8")
     print_lines(output)
     installed_version = get_agent_version(cont)
-    if agent_version:
-        assert installed_version == agent_version, "installed agent version is '%s', expected '%s'" % (
-            installed_version,
-            agent_version,
-        )
-    return installed_version
+    assert installed_version == agent_version, "installed agent version is '%s', expected '%s'" % (
+        installed_version,
+        agent_version,
+    )
+    assert is_agent_running_as_non_root(cont), "Agent is not running as non-root user"
+    backend.reset_datapoints()
```

Comment:
Same thing here.

Reply:
Ok, I see what you are trying to do with it.  Do the reset in the test code instead of here so it doesn't mix concerns so much.

Reply:
> Ok, I see what you are trying to do with it. Do the reset in the test code instead of here so it doesn't mix concerns so much.

Will move out of the function and into the test code.